authors/apps/articles/views.py

```diff
@@ -0,0 +1,94 @@
+import json
+
+from django.conf import settings
+from django.contrib.sites.shortcuts import get_current_site
+from django.core.mail import send_mail
+from django.shortcuts import get_object_or_404
+from django.urls import reverse
+from django.utils.encoding import force_bytes
+from django.utils.encoding import force_text
+from django.utils.http import urlsafe_base64_encode, urlsafe_base64_decode
+from rest_framework import status
+from rest_framework.generics import ListCreateAPIView, RetrieveUpdateDestroyAPIView
+from rest_framework.permissions import AllowAny, IsAuthenticated, IsAuthenticatedOrReadOnly
+from rest_framework.response import Response
+
+from config.settings import default
+from .models import Article
+from .renderers import ArticleJSONRenderer
+from .serializers import (
+     ArticleSerializer
+)
+
+from authors.apps.profiles.models import Profile
+from authors.apps.authentication.models import User
+
+
+class ArticleView(ListCreateAPIView):
+    """creating, viewing , deleting and updating articles"""
+
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+    renderer_classes = (ArticleJSONRenderer, )
+
+    def post(self, request):
+        serializer_context = {'author': request.user.profile}
+        article = request.data
+
+        # Create an article from the above data
+        serializer = ArticleSerializer(data=article, context=serializer_context)
```

Comment:
@frankopkusianwar , add the serializer class to the class variables, and call it in the methods using `self`